packages/manager/modules/dbaas-logs/src/logs/detail/streams/logs-streams.service.js

```diff
@@ -425,6 +425,30 @@ export default class LogsStreamsService {
     });
   }
 
+  /**
+   * copies stream ID to clipboard.
+   *
+   * @param {any} stream
+   * @memberof LogsStreamsService
+   */
+  copyStreamId(stream) {
+    const error = this.CucControllerHelper.constructor.copyToClipboard(
```

Comment:
the copyToClipboard function doesn't work on safari (cf DTRSD-126045 issue). 

In the CucControllerHelper.copyToClipboard function, the function needs to be enriched with the navigator.clipboard API usage and keep the current one as a fallback if navigator api is not supported by the user's browser : 

```   
static copyToClipboard(messageToCopy) {
 // check if API available on browser
  if (navigator.clipboard && navigator.clipboard.writeText) {
    return navigator.clipboard.writeText(messageToCopy)
      .then(() => '')
      .catch(err => err);
  } else {
  // current usage
    try {
      const dummy = document.createElement('input');
      document.body.appendChild(dummy);
      dummy.setAttribute('id', 'dummy_id');
      dummy.setAttribute('value', messageToCopy);
      dummy.select();
      document.execCommand('copy');
      document.body.removeChild(dummy);
    } catch (err) {
      return err;
    }
    return '';
  }
}
```

Reply:
I applied this change with a small diff, to always return an empty string in case of success and a not-empty string in case of failure (and not a promise). 
Otherwise it would have break current usage of this function